src/app/components/shared/typeahead-input/typeahead-input.component.ts

```diff
@@ -0,0 +1,112 @@
+  OperatorFunction,
+  switchMap,
+} from "rxjs";
+import { defaultDebounceTime } from "src/app/app.helper";
+
+@Component({
+  selector: "baw-typeahead-input[optionsCallback]",
+  templateUrl: "typeahead-input.component.html",
+  styleUrls: ["typeahead-input.component.scss"],
+})
+export class TypeaheadInputComponent {
+  public constructor() {}
+
+  /** A behavior subject that emits the current models selected/active */
+  @Input() public modelChange: BehaviorSubject<AbstractModel[] | string[]> =
+    new BehaviorSubject([]);
+  /** The options callback is typically linked to a service as it should return a list observable of options that the user could select */
+  @Input() public optionsCallback: (
+    text: string
+  ) => Observable<AbstractModel[] | string[]>;
+  /** Describes how to convert an object model into a human readable form for use in the pills and typeahead dropdown */
+  @Input() public formatter: (item: AbstractModel) => string = (
+    item: AbstractModel
+  ): string => item.toString();
+  /** Whether the typeahead input should allow multiple inputs in pill form */
+  @Input() public multipleInputs = false;
+  /** Text to show above the input field. Usually a one 1-2 word descriptor. */
+  @Input() public label = "";
+  /** Placeholder text that is shown when the input field is empty.
+   * Note: This value is not emitted at any point
+   */
+  @Input() public inputPlaceholder = "";
+
+  @ViewChild("typeaheadInputRef")
+  public typeaheadInput: ElementRef<HTMLInputElement>;
+  public inputModel: string;
+
+  /** if multiple items are enabled, they will be added to the activeItems */
+  public activeItems: AbstractModel[] = [];
+  /** if the user clicks on a pill, it will be the selected item */
+  public selectedItem: AbstractModel;
+
+  public findOptions: OperatorFunction<string, readonly unknown[]> = (
+    text$: Observable<string>
+  ) => {
+    const maximumResults = 10;
+
+    return text$.pipe(
+      debounceTime(defaultDebounceTime),
+      distinctUntilChanged(),
+      switchMap((term: string) =>
+        this.optionsCallback(term).pipe(
+          map((items: AbstractModel[]) =>
+            items
+              .filter(
+                (item: AbstractModel) =>
+                  this.formatter(item)
+                    .toLowerCase()
+                    .includes(term.toLowerCase()) &&
+                  !this.activeItems.some(
+                    (activeItem: AbstractModel) =>
+                      activeItem.toString() === item.toString()
+                  )
+              )
+              .slice(0, maximumResults)
+          )
+        )
+      )
+    );
+  };
+
+  public onItemSelected($event: NgbTypeaheadSelectItemEvent<AbstractModel>) {
+    $event.preventDefault();
+    const selectedItem: AbstractModel = $event.item;
+
+    if (this.multipleInputs) {
+      this.activeItems.push(selectedItem);
+      this.modelChange.next(this.activeItems);
+
+      this.typeaheadInput.nativeElement.value = "";
```

Comment:
seems like you want to use bootstrap's own api for this?

Reply:
Not quite sure what you mean here...

However, I was able to fix the `value = ""` by using a two-way data-binding, and setting the model to `null`

Reply:
by bypassing the typeahead's api and setting the dom element's value you're essentially bypassing any mechanisms and change detection the typeahead component has. You might as well be doing a document.querySelector here and mutate the dom directly.

if you had to reset the value with code you should be using `this.typeaheadInput.writeValue` 

See https://angular.io/api/forms/ControlValueAccessor
See https://github.com/ng-bootstrap/ng-bootstrap/blob/master/src/typeahead/typeahead.ts